swap/swap.go

```diff
@@ -466,56 +303,56 @@ func (s *Swap) Balances() (map[enode.ID]int64, error) {
 	return balances, nil
 }
 
-// loadLastSentCheque loads the last cheque for a peer from the state store (persisted)
-// To be called with mutex already held
-// Caller must be careful that the same cheque isn't concurrently read and written by multiple routines
-func (s *Swap) loadLastSentCheque(peer enode.ID) (err error) {
-	//only load if the current instance doesn't already have this peer's
-	//last cheque in memory
+// loadLastReceivedCheque loads the last received cheque for the peer from the store
+func (s *Swap) loadLastReceivedCheque(p enode.ID) (*Cheque, error) {
 	var cheque *Cheque
-	if _, ok := s.getCheque(peer); !ok {
-		err = s.store.Get(sentChequeKey(peer), &cheque)
-		if err == nil {
-			s.setCheque(peer, cheque)
-		}
+	error := s.store.Get(receivedChequeKey(p), &cheque)
```

Comment:
You can put ``error` and `cheque` to the function initialization

Reply:
Not sure what you're proposing here.

Reply:
I meant that you can change 
```golang
func (s *Swap) loadLastReceivedCheque(p enode.ID) (*Cheque, error) {
	var cheque *Cheque
	error := s.store.Get(receivedChequeKey(p), &cheque)
...
```
to 
```golang
func (s *Swap) loadLastReceivedCheque(p enode.ID) (ch *Cheque, err error) {
...
```
This is what I would prefer, but it is merely style, ofcourse. I would be curious to find out if there are any guidelines here in Swarm. You prefer it in the way you did, I assume?

Reply:
Looking at code from the `stream` package it seems both variants are used throughout the code. Doing it that way is fine from my side as long we still return the values explicitly.

Reply:
Changed